i_1061 Prompt for credentials for pc2submit if none found

johnbrvc opened the pull request:
### Description of what the PR does
Add optional code (controlled by environment variable **SUBMITCREDENTIALS** == '_yes_') to prompt the user for their contest credentials and add them to their `~/.netrc` file.  Added appropriate code ask for and confirm the user's credentials.
CI: Code cleanup - create constants for environment variables (there were already several environment variables hard-coded).
CI: Add `-t #` (`--team_id #`) to allow specification of the `team_id `to submit on behalf of
CI: Add environment variable **SUBMITOMITEXTRAHELP**=='_yes_' to not include help for the `-c`, `-a`, `-u` and `-t `options as these are typically not used by most users.

### Issue which the PR addresses
Related to #1061.  See also #1063 

### Environment in which the PR was developed (OS,IDE, Java version, etc.)
Windows 11
java version "1.8.0_321"
Java(TM) SE Runtime Environment (build 1.8.0_321-b07)
Java HotSpot(TM) 64-Bit Server VM (build 25.321-b07, mixed mode)

### Precise steps for _testing_ the PR (i.e., how to demonstrate that it works correctly)

1. Remove any "**localhost**" entry in `~/.netrc`
2. Start a contest (like **sumithello**)
3. Start a feeder and start it feeding 2023-06 API
4. Set/Unset the `SUBMITCREDENTIALS `environment to _yes_ or _no_ (to test various combinations)
5. Make a submission using `pc2submit`, eg. `pc2submit -p a -y --team_id 1 samps/src/hello.cpp`
6. You should be prompted to enter credentials if `SUBMITCREDENTIALS` was set to '_yes_', otherwise it should just fail with a network exception of some sort.
7. You can repeat the test using different values of `SUBMITCREDENTIALS`, user and password, eg. team1/team1, administrator1/administrator1, or wrong credentials.  Be sure to remove the entry for localhost from the `~/.netrc `if you want to clear out the credentials for a new test.

johnbrvc committed:
i_1062 Prompt for credentials for pc2submit if none found

Add optional code (controlled by environment variable SUBMITCREDENTIALS == 'yes') to prompt the user for their contest credentials and add them to their ~/.netrc file.  Added appropriate code ask for and confirm the user's credentials.
CI: Code cleanup - create constants for environment variables (there were already several environment variables hard-coded).
CI: Add -t # (--team_id #) to allow specification of the team_id to submit on behalf of
CI: Add environment variable SUBMITOMITEXTRAHELP=='yes' to not include help for the -c, -a, -u and -t options as these are typically not used by most users.

SamanwaySadhu reviewed (approved):
#1063 also mentions that it fixes #1061. Maybe one of the two should mention Partial fix. Other than that it looks good.

johnbrvc commented:
> #1063 also mentions that it fixes #1061. Maybe one of the two should mention Partial fix. Other than that it looks good.

#1063 is more of a reference to #1061.  Ill fix that one.

johnbrvc committed:
i_1061 Remote dateutil dependency

The python3 dateutil package is not included on the default contest image.  Therefore, change the code to use datetime, which is included.  We also have to check for an abbreviated form of timezone and fix it, which datetime does not support (but dateutil did).

johnbrvc committed:
i_1061 Bug fixes for pc2submit (python version)

Fix bugs found when testing contest image.
Can't use os.getlogin() in GUI environment - login name can't be determined.  So, just user the USER environment variable as default.
Re-arrange code for better exception handling on missing .netrc file.

johnbrvc committed:
i_1062 Clean up user interface, minor bug fixes

Made UI more user friendly for credentials.
Fix "c" language specification.
Fix timeformat if zulu time in response.

johnbrvc committed:
Fixed bug with failed credentials
